Core/Src/fault.c

```diff
@@ -1,3 +1,45 @@
 #include "fault.h"
 
 osMessageQueueId_t fault_handle_queue;
+
+//Function to determine priority base on severity
+void priority(void) {
+    osThreadId_t id;
+    MSGQUEUE_OBJ_t msg;
+    osStatus_t status;
+    // Wait until a message is in the queue, send messages when they are in the queue
+	for(;;) {
+        status = osMessageQueueGet(fault_handle_queue, &msg, NULL, 0U);   // wait for message
+        id = osThreadGetId(); // Obtain ID of current running thread
+        if (status == osOK) {
+        
+        // process data
+        switch (fault_sev_t) {
```

Comment:
You want to access the severity field of the `fault_t` or whatever data type you get from the queue. The idea is that we are popping off a fault and then routing to a case for the severity